### PR TITLE
sql-3: As a developer I want to have login and registration routes

### PR DESCRIPTION
**Pull request by OlegKarapysh**
added jwt authentication

### CODE REVIEW

File: backend/Squirrel.Core/Squirrel.Core.BLL/Services/AuthService.cs [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    }
+
+    public async Task<RefreshedAccessTokenDto> Login(UserLoginDto userLoginDto)
+    {
+        // TODO: Find user in database by his email and get user info. Exception when not found or invalid credentials.
+        // Dummy user info.
+        var userId = 0;
+        var username = "username";
+        
+        var token = await GenerateNewAccessToken(userId, username, userLoginDto.Email);
```
Comment (Anatolii-Ivanov) [inline backend/Squirrel.Core/Squirrel.Core.BLL/Services/AuthService.cs]: No need to assign variable
Just return await ...

Comment (Anatolii-Ivanov) [reply backend/Squirrel.Core/Squirrel.Core.BLL/Services/AuthService.cs]: Everywhere

File: backend/Squirrel.Core/Squirrel.Core.BLL/Services/AuthService.cs [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        
+        return token;
+    }
+
+    public async Task<RefreshedAccessTokenDto> Register(UserRegisterDto userRegisterDto)
+    {
+        if (userRegisterDto.Password != userRegisterDto.ConfirmPassword)
```
Comment (Anatolii-Ivanov) [inline backend/Squirrel.Core/Squirrel.Core.BLL/Services/AuthService.cs]: This check should be on frontend, we won't send it to FE

File: backend/Squirrel.Core/Squirrel.Core.Common/JWT/AccessToken.cs [hunk 3]
```diff
@@ -0,0 +1,13 @@
+﻿namespace Squirrel.Core.Common.JWT;
+
+public sealed class AccessToken
+{
+    public string Token { get; }
+    public int ExpiresIn { get; }
```
Comment (Anatolii-Ivanov) [inline backend/Squirrel.Core/Squirrel.Core.Common/JWT/AccessToken.cs]: This part is stored in token itself, so no need to send additional data (Which are valid only for millisecond)

File: backend/Squirrel.Core/Squirrel.Core.Common/Squirrel.Core.Common.csproj [hunk 4]
```diff
@@ -6,4 +6,17 @@
...
+    <PackageReference Include="Microsoft.AspNetCore.Cryptography.KeyDerivation" Version="7.0.10" />
+    <PackageReference Include="Microsoft.IdentityModel.Tokens" Version="6.10.0" />
+    <PackageReference Include="Newtonsoft.Json" Version="13.0.1" />
+    <PackageReference Include="System.IdentityModel.Tokens.Jwt" Version="6.10.0" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <Reference Include="Microsoft.Extensions.Options">
+      <HintPath>..\..\..\..\..\..\..\..\..\..\Program Files\dotnet\shared\Microsoft.AspNetCore.App\6.0.11\Microsoft.Extensions.Options.dll</HintPath>
+    </Reference>
```
Comment (Anatolii-Ivanov) [inline backend/Squirrel.Core/Squirrel.Core.Common/Squirrel.Core.Common.csproj]: What is it:)

File: backend/Squirrel.Core/Squirrel.Core.WebAPI/Controllers/SampleController.cs [hunk 5]
```diff
@@ -1,11 +1,13 @@
...
+﻿using Microsoft.AspNetCore.Authorization;
+using Squirrel.Core.BLL.Interfaces;
 using Squirrel.Core.Common.DTO;
 using Microsoft.AspNetCore.Mvc;
 
 namespace Squirrel.Core.WebAPI.Controllers
 {
     [ApiController]
     [Route("[controller]")]
+    [Authorize] // For authentication tests.
```
Comment (Anatolii-Ivanov) [inline backend/Squirrel.Core/Squirrel.Core.WebAPI/Controllers/SampleController.cs]: You don't need to commit this line :)

File: backend/Squirrel.Core/Squirrel.Core.WebAPI/Controllers/AuthController.cs [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    [HttpPost("login")]
+    public async Task<ActionResult<RefreshedAccessTokenDto>> Login([FromBody] UserLoginDto userLoginData)
+    {
+        return Ok(await _authService.Login(userLoginData));
+    }
+    
+    /// <summary>
+    /// Registers new user and generates tokens
+    /// </summary>
+    /// /// /// <remarks>
```
Comment (Anatolii-Ivanov) [inline backend/Squirrel.Core/Squirrel.Core.WebAPI/Controllers/AuthController.cs]: I guess you don't need so many / here) The same with previous one

File: backend/Squirrel.Core/Squirrel.Core.Common/JWT/JwtFactory.cs [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+                throw new SecurityTokenException("Invalid token");
+            }
+
+            return principal;
+        }
+        catch (Exception)
+        {
+            return null;
+        }
+    }
```
Comment (Anatolii-Ivanov) [inline backend/Squirrel.Core/Squirrel.Core.Common/JWT/JwtFactory.cs]: This is done by ASP.NET Core itself, are you sure we need this?

**Review by Anatolii-Ivanov — CHANGES_REQUESTED** [review]

**Review by Anatolii-Ivanov — DISMISSED** [review]

**Review by VladTkach — DISMISSED** [review]

**Review by VladTkach — DISMISSED** [review]

**Review by Anatolii-Ivanov — APPROVED** [review]

**Review by VladTkach — APPROVED** [review]

**Pull request merged**